### DISCUSSION TITLE
2 things I can't figure out - new page template and bookmarks

### DISCUSSION THREAD

**Discussion by brucewestfall** [zim-desktop-wiki/zim-desktop-wiki#2345] [General]
So the first one about new page template:  I have a default page template that works exactly like it should.  I can generate new pages for jobs with a script and some python.  works great so far!
I want to create a link on a newly created page (using the default) that will make a link to a subpage ( +nameofpage ) but i want it to use a different template.  Right now i can call zim with python and specify the template, but I also want this secondary page.  I'm guessing there is a format to use for a link like that and i'm just missing it.  I just want a link made because only certain jobs will need this.  Hope that makes sense.

Other thing:
Just how and where are bookmarks located?  Would be nice to sync them, but can't find the file location.  Using Windows 11 and Linux mint, so any answer should start me off in the right direction.

Oh yea, and i absolutely LOVE zim.  Other apps do some things better, but I've found nothing that can duplicate zim AND so easy to learn!!

**Comment by introt** [comment]
Hi!

> Right now i can call zim with python and specify the template, but I also want this secondary page.

I'm not quite sure I got this right, but could you solve this by adding a second call to your Python script that creates `primary page:secondary page` using the other template? Please elaborate with a (simplified) example of the two pages and your current call to create the first one.

> Just how and where are bookmarks located?

Assuming you mean the BookmarksBar plugin, they're stored in the user interface state file. Don't ask me why - that's the way it has been the plugin was introduced [in 2015](https://github.com/zim-desktop-wiki/zim-desktop-wiki/commit/2c6ec4c1d4c14766f8abde17aa1b7b429e062d3e).

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/4110a13c391c24aa92c8b4edfc77bb79563d4368/zim/plugins/bookmarksbar.py#L200

On Linux, you can usually find the file at `$HOME/.cache/zim/*/state.conf`. I'm not sure whether you'd want to sync the whole file between Windows and Linux as it contains window sizes and such, but if you're handy with Python feel free to pick out the line containing the bookmarks - or better yet, contribute an alternative bookmark saving mechanism. The current one is rather simple:

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/4110a13c391c24aa92c8b4edfc77bb79563d4368/zim/plugins/bookmarksbar.py#L498-L501

One could swap out `uistate` with its `notebook.zim` equivalent pretty easily - see the Journal plugin for example usage of `plugin_notebook_properties`. I've opened #2346 to track this.
